geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -31,58 +34,77 @@ trait IngestCommand[DS <: DataStore] extends DataStoreCommand[DS] {
+      createShpIngest().run()
     } else {
-      // validate arguments
-      if (params.config == null) {
-        throw new ParameterException("Converter Config argument is required")
-      }
-      val converterConfig = CLArgResolver.getConfig(params.config)
-
-      val sft = if (params.spec != null) {
-        CLArgResolver.getSft(params.spec, params.featureName)
-      } else if (params.featureName != null) {
-        Try(withDataStore(_.getSchema(params.featureName))).filter(_ != null).getOrElse {
-          throw new ParameterException(s"SimpleFeatureType '${params.featureName}' does not currently exist, " +
-              "please provide specification argument")
+      // try to load the sft, first check for an existing schema, then load from the params/environment
+      var sft: SimpleFeatureType =
+        Option(params.featureName).flatMap(n => Try(withDataStore(_.getSchema(n))).filter(_ != null).toOption)
+          .orElse(Option(params.spec).flatMap(s => Option(CLArgResolver.getSft(s, params.featureName))))
+          .orNull
+
+      var converter: Config = Option(params.config).map(CLArgResolver.getConfig).orNull
+
+      if (converter == null && params.files.nonEmpty) {
+        // if there is no converter passed in, try to infer the schema from the input files themselves
+        Command.user.info("No converter defined - will attempt to detect schema from input files")
+        val file = params.files.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
+          throw new ParameterException(s"<files> '${params.files.mkString(",")}' did not evaluate to anything" +
+              "that could be read")
         }
-      } else {
+        val inferred = SimpleFeatureConverter.infer(() => file.open, Option(sft)).getOrElse {
+          throw new ParameterException("Could not determine converter from inputs - please specify a converter")
+        }
+
+        if (sft == null) {
+          val typeName = Option(params.featureName).getOrElse {
+            val existing = withDataStore(_.getTypeNames)
+            val fileName = Option(FilenameUtils.getBaseName(file.path))
+            val base = fileName.map(_.trim.replaceAll("[^A-Za-z0-9]+", "_")).filterNot(_.isEmpty).getOrElse("geomesa")
+            var name = base
+            var i = 0
+            while (existing.contains(name)) {
+              name = s"${base}_$i"
+              i += 1
+            }
+            name
+          }
+          sft = SimpleFeatureTypes.renameSft(inferred._1, typeName)
+          Command.user.info(s"Inferred schema: $typeName identified ${SimpleFeatureTypes.encodeType(sft)}")
+        }
+        Command.user.info("Inferred converter:\n" +
+            inferred._2.root().render(ConfigRenderOptions.concise().setFormatted(true)))
+        if (Prompt.confirm("Use inferred converter (y/n)? ")) {
+          Command.user.info("For consistency, please use this converter definition for any future ingest")
+          converter = inferred._2
```

Comment:
Does the user have a chance to save the converter?  Or would they manually have to do so?

Reply:
they would have to manually copy it out. i'll add a ticket to write it out for them

Reply:
https://geomesa.atlassian.net/browse/GEOMESA-2327